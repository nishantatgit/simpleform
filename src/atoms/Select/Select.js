import React, { useState } from 'react';

export default function Select(props) {
  const { options, label, id, onChange, name, value, onBlur } = props;
  //const [selectedValue, setSelectedValue] = useState('select');
  // const onChangeHandler = e => {
  //  // setSelectedValue(e.target.value);
  //  // console.log('selectedValue', e.target.value);
  //   onChange && onChange();
  // };
  // function onBlur() {
  //   console.log('on blur called');
  //   handleBlur();
  // }

  return (
    <>
      <label htmlFor={id}>{label}</label>
      <select
        id={id}
        onChange={onChange}
        value={value}
        name={name}
        onBlur={onBlur}
      >
        <option selected value="">
          Select
        </option>
        {options &&
          options.map(option => (
            <option value={option.value}>{option.label}</option>
          ))}
      </select>
    </>
  );
}
