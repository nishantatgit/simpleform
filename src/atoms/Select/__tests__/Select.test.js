import React from 'react';
import { render, shallow } from 'enzyme';
import Select from '../Select';

describe('<Select />', () => {
  const selectOptions = [
    {
      value: 1,
      label: 'Mr.',
    },
    {
      value: 2,
      label: 'Mrs.',
    },
  ];
  const props = {
    id: 'name-select',
    name: 'titleSelect',
    label: 'Select title',
    selectOptions,
  };

  it('should render select correctly', () => {
    const selectEl = render(<Select {...props} />);
    expect(selectEl).toMatchSnapshot();
  });

  it('calls onChange only once per change', () => {
    const addOnProps = {
      ...props,
      onChange: jest.fn(),
    };
    const selectElWithChange = shallow(<Select {...addOnProps} />);
    selectElWithChange.find('select').simulate('change');
    expect(addOnProps.onChange).toHaveBeenCalledTimes(1);
  });

  it('calls onBlur only once on blur', () => {
    const addOnProps = {
      ...props,
      onBlur: jest.fn(),
    };
    const selectElWithBlur = shallow(<Select {...addOnProps} />);
    selectElWithBlur.find('select').simulate('blur');
    expect(addOnProps.onBlur).toHaveBeenCalledTimes(1);
  });
});
