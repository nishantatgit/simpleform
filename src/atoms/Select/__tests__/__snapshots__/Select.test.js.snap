// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render select correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "for": "name-select",
    },
    "children": Array [
      Object {
        "data": "Select title",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "label",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": Object {
      "attribs": Object {
        "id": "name-select",
        "name": "titleSelect",
      },
      "children": Array [
        Object {
          "attribs": Object {
            "selected": "",
            "value": "",
          },
          "children": Array [
            Object {
              "data": "Select",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "option",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "selected": undefined,
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "selected": undefined,
            "value": undefined,
          },
        },
      ],
      "name": "select",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": [Circular],
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "id": undefined,
        "name": undefined,
      },
      "x-attribsPrefix": Object {
        "id": undefined,
        "name": undefined,
      },
    },
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
        Object {
          "attribs": Object {
            "id": "name-select",
            "name": "titleSelect",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "selected": "",
                "value": "",
              },
              "children": Array [
                Object {
                  "data": "Select",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "option",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "selected": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "selected": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "select",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": null,
          "prev": [Circular],
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "id": undefined,
            "name": undefined,
          },
          "x-attribsPrefix": Object {
            "id": undefined,
            "name": undefined,
          },
        },
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "for": undefined,
    },
    "x-attribsPrefix": Object {
      "for": undefined,
    },
  },
  "1": Object {
    "attribs": Object {
      "id": "name-select",
      "name": "titleSelect",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "selected": "",
          "value": "",
        },
        "children": Array [
          Object {
            "data": "Select",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "option",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "selected": undefined,
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "selected": undefined,
          "value": undefined,
        },
      },
    ],
    "name": "select",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": Object {
      "attribs": Object {
        "for": "name-select",
      },
      "children": Array [
        Object {
          "data": "Select title",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
      ],
      "name": "label",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": [Circular],
      "parent": null,
      "prev": null,
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "for": undefined,
      },
      "x-attribsPrefix": Object {
        "for": undefined,
      },
    },
    "root": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "attribs": Object {
            "for": "name-select",
          },
          "children": Array [
            Object {
              "data": "Select title",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "label",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": null,
          "prev": null,
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "for": undefined,
          },
          "x-attribsPrefix": Object {
            "for": undefined,
          },
        },
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "id": undefined,
      "name": undefined,
    },
    "x-attribsPrefix": Object {
      "id": undefined,
      "name": undefined,
    },
  },
  "_root": [Circular],
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
